dfs/backup.go

```diff
@@ -34,30 +37,18 @@ const (
 
 // Backup writes all application data into an export stream
 func (dfs *DistributedFilesystem) Backup(data BackupInfo, w io.Writer) error {
-	tarfile := tar.NewWriter(w)
-	defer tarfile.Close()
-	buffer := bytes.NewBufferString("")
-	spool, err := utils.NewSpool(dfs.tmp)
-	if err != nil {
-		glog.Errorf("Could not create spool: %s", err)
-		return err
-	}
-	defer spool.Remove()
-	// write the backup metadata
-	glog.Infof("Writing backup metadata")
-	if err := json.NewEncoder(buffer).Encode(data); err != nil {
-		glog.Errorf("Could not encode backup metadata: %s", err)
-		return err
-	}
-	header := &tar.Header{Name: BackupMetadataFile, Size: int64(buffer.Len())}
-	if err := tarfile.WriteHeader(header); err != nil {
-		glog.Errorf("Could not create metadata header for backup: %s", err)
-		return err
-	}
-	if _, err := buffer.WriteTo(tarfile); err != nil {
-		glog.Errorf("Could not write backup metadata: %s", err)
+
+	// A set of pipes that output tar streams
+	var tarpipes []pipe.Pipe
+
+	// write the backup metadata. This is too tiny to parallelize
+	metadataBuffer := &bytes.Buffer{}
+	if err := dfs.writeBackupMetadata(data, metadataBuffer); err != nil {
+		glog.V(2).Infof("Unable to write backup metadata: %s", err)
```

Comment:
That's a good question, and I don't know the right answer. The reason I made it not an errorf is because it ends up being double-logged then, once here, and once when the error gets returned up the stack. I think the right approach is debug log it at this low level, and error log it at the high level.